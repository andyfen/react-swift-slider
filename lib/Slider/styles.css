.container {
  position: relative;
}
.row {
  display: flex;
  flex-direction: row;
}
.slides {
    position: relative;
    padding: 0px;
    margin: 0px;
    list-style-type: none;
}

.dots {
  list-style: none;
  margin: 1.7rem 0;
  padding: 0;
  display: flex;
  flex-direction: row;
  justify-content: center;
}
