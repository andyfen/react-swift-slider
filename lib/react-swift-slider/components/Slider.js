'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true,
});
exports['default'] = void 0;

var _react = _interopRequireWildcard(require('react'));

var _propTypes = _interopRequireDefault(require('prop-types'));

var _styledComponents = _interopRequireDefault(require('styled-components'));

var _Slide = _interopRequireDefault(require('./Slide'));

var _Control = _interopRequireWildcard(require('./Control'));

var _Dot = _interopRequireDefault(require('./Dot'));

var _useInterval = require('../hooks/use-interval');

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj;
  } else {
    var newObj = {};
    if (obj != null) {
      for (var key in obj) {
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
          var desc =
            Object.defineProperty && Object.getOwnPropertyDescriptor
              ? Object.getOwnPropertyDescriptor(obj, key)
              : {};
          if (desc.get || desc.set) {
            Object.defineProperty(newObj, key, desc);
          } else {
            newObj[key] = obj[key];
          }
        }
      }
    }
    newObj['default'] = obj;
    return newObj;
  }
}

function _slicedToArray(arr, i) {
  return (
    _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest()
  );
}

function _nonIterableRest() {
  throw new TypeError('Invalid attempt to destructure non-iterable instance');
}

function _iterableToArrayLimit(arr, i) {
  var _arr = [];
  var _n = true;
  var _d = false;
  var _e = undefined;
  try {
    for (
      var _i = arr[Symbol.iterator](), _s;
      !(_n = (_s = _i.next()).done);
      _n = true
    ) {
      _arr.push(_s.value);
      if (i && _arr.length === i) break;
    }
  } catch (err) {
    _d = true;
    _e = err;
  } finally {
    try {
      if (!_n && _i['return'] != null) _i['return']();
    } finally {
      if (_d) throw _e;
    }
  }
  return _arr;
}

function _arrayWithHoles(arr) {
  if (Array.isArray(arr)) return arr;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral([
    '\n  list-style: none;\n  margin: 1.7rem 0;\n  padding: 0;\n  display: flex;\n  flex-direction: row;\n  justify-content: center;\n  @media (max-width: 600px) {\n    margin: 1rem 0;\n  }\n',
  ]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral([
    '\n  position: relative;\n  padding: 0;\n  margin: 0;\n  list-style: none;\n  height: ',
    'px;\n  @media (max-width: 600px) {\n    height: 250px;\n  }\n',
  ]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral([
    '\n  position: relative;\n  height: ',
    'px;\n  @media (max-width: 600px) {\n    height: 250px;\n  }\n',
  ]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }
  return Object.freeze(
    Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } }),
  );
}

var SwiftSliderContainer = _styledComponents['default'].div(
  _templateObject(),
  function(props) {
    return props.height || '450';
  },
);

var SwiftSliderSlides = _styledComponents['default'].ul(
  _templateObject2(),
  function(props) {
    return props.height || '450';
  },
);

var SwiftSliderDots = _styledComponents['default'].ul(_templateObject3());

function Slider(_ref) {
  var data = _ref.data,
    height = _ref.height,
    activeDotColor = _ref.activeDotColor,
    interval = _ref.interval,
    dotColor = _ref.dotColor,
    showDots = _ref.showDots,
    enableNextAndPrev = _ref.enableNextAndPrev;

  var _useState = (0, _react.useState)(0),
    _useState2 = _slicedToArray(_useState, 2),
    currentSlide = _useState2[0],
    setCurrentSlide = _useState2[1];

  var _useState3 = (0, _react.useState)(interval),
    _useState4 = _slicedToArray(_useState3, 2),
    currentInterval = _useState4[0],
    setCurrentInterval = _useState4[1];

  (0, _useInterval.useInterval)(function() {
    nextSlide();
  }, currentInterval);
  (0, _react.useEffect)(
    function() {
      data.forEach(function(img) {
        var image = new Image();
        image.src = img.src;
      });
    },
    [data],
  );

  var nextSlide = function nextSlide() {
    if (currentSlide === data.length - 1) {
      return setCurrentSlide(0);
    }

    setCurrentSlide(currentSlide + 1);
  };

  var prevSlide = function prevSlide() {
    if (currentSlide === 0) {
      return setCurrentSlide(data.length - 1);
    }

    setCurrentSlide(currentSlide - 1);
    setCurrentInterval(currentInterval + 1);
  };

  var goToSlide = function goToSlide(idx) {
    setCurrentSlide(idx);
    setCurrentInterval(currentInterval - 1);
  };

  return _react['default'].createElement(
    SwiftSliderContainer,
    null,
    _react['default'].createElement(
      SwiftSliderSlides,
      {
        height: height,
      },
      data.map(function(item, i) {
        return _react['default'].createElement(_Slide['default'], {
          active: i === currentSlide,
          src: item.src,
          key: item.id,
        });
      }),
    ),
    showDots
      ? _react['default'].createElement(
          SwiftSliderDots,
          null,
          data.map(function(item, i) {
            return _react['default'].createElement(_Dot['default'], {
              activeDotColor: activeDotColor,
              dotColor: dotColor,
              key: i,
              active: i === currentSlide,
              onClick: goToSlide,
              idx: i,
            });
          }),
        )
      : '',
    enableNextAndPrev
      ? _react['default'].createElement(_Control['default'], {
          onPressPrev: prevSlide,
          direction: _Control.DIRECTION.prev,
        })
      : '',
    enableNextAndPrev
      ? _react['default'].createElement(_Control['default'], {
          onPressNext: nextSlide,
          direction: _Control.DIRECTION.next,
        })
      : '',
  );
}

Slider.propTypes = {
  data: _propTypes['default'].array.isRequired,
  height: _propTypes['default'].number,
  interval: _propTypes['default'].number,
  activeDotColor: _propTypes['default'].string,
  dotColor: _propTypes['default'].string,
  showDots: _propTypes['default'].bool,
  enableNextAndPrev: _propTypes['default'].bool,
};
Slider.defaultProps = {
  data: [],
  height: 450,
  activeDotColor: '#e8784e',
  interval: 5000,
  dotColor: '#909192',
  showDots: true,
  enableNextAndPrev: true,
};
var _default = Slider;
exports['default'] = _default;
