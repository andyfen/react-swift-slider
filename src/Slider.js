import React from 'react';
import { StyleSheet, css } from 'aphrodite';

import Slide from './Slide';
import Control from './Control';
import Dot from './Dot';

export default class Slider extends React.Component {
  static propTypes = {
    data: React.PropTypes.array.isRequired,
    height: React.PropTypes.number,
    interval: React.PropTypes.number,
    activeDotColor: React.PropTypes.string,
    dotColor: React.PropTypes.string
  };
  static defaultProps = {
    height: 450,
    activeDotColor: '#e8784e',
    interval: 5000,
    dotColor: '#909192'
  };
  constructor(props){
    super(props);
    this.state = {
      currentSlide: 0,
    }
    this.slideInterval = setInterval(this.nextSlide, this.props.interval);
  }
  resetInterval = () => {
    clearInterval(this.slideInterval)
    this.slideInterval = setInterval(this.nextSlide, this.props.interval);
  }
  nextSlide = () => {
    if(this.state.currentSlide === this.props.data.length-1){
      this.setState({
        currentSlide: 0
      })
      return;
    }
    this.setState({
      currentSlide: this.state.currentSlide +1
    })
    this.resetInterval();
  }

  prevSlide = () => {
    if(this.state.currentSlide === 0){
      this.setState({
        currentSlide: this.props.data.length - 1
      })
      return;
    }
    this.setState({
      currentSlide: this.state.currentSlide - 1
    })
    this.resetInterval();
  }
  goToSlide = (idx) => {
    this.setState({
      currentSlide: idx
    })
    this.resetInterval();
  }
  render(){
    const { data, height, activeDotColor, dotColor } = this.props;
    return(
      <div className={css(styles.container)}>
        <ul className={css(styles.slides)} style={{height: `${height}px !important`}}>
          {data.map((item, i) =>
            <Slide active={i === this.state.currentSlide}
              src={item.src}
              key={item.id}/>
          )}
        </ul>
        <ul className={css(styles.dots)} >
          {data.map((item, i) =>
            <Dot
              activeDotColor={activeDotColor}
              dotColor={dotColor}
              key={i} active={i === this.state.currentSlide}
              onClick={this.goToSlide}
              idx={i}
            />
          )}
        </ul>
        <Control
          onPressPrev={this.prevSlide}
          direction="prev"/>
        <Control
          onPressNext={this.nextSlide}
          direction="next"/>
      </div>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    position: 'relative',
    height: '450px',
    '@media (max-width: 600px)': {
      height: '250px',
    }
  },
  row: {
    display: 'flex',
    flexDirection: 'row'
  },
  slides: {
    height: '450px',
    position: 'relative',
    padding: '0px',
    margin: '0px',
    listStyle: 'none',
    '@media (max-width: 600px)': {
      height: '250px',
    }
  },
  dots: {
    listStyle: 'none',
    margin: '1.7rem 0',
    padding: 0,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    '@media (max-width: 600px)': {
      margin: '1rem 0',
    }
  }
})
