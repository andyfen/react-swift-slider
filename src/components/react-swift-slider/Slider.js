import React, { Component } from "react";
import PropTypes from "prop-types";
import "./assets/sass/react-swift-slider.css";

import Slide from "./Slide";
import Control from "./Control";
import Dot from "./Dot";

export default class Slider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentSlide: 0
    };
    this.slideInterval = setInterval(this.nextSlide, this.props.interval);
  }
  resetInterval = () => {
    clearInterval(this.slideInterval);
    this.slideInterval = setInterval(this.nextSlide, this.props.interval);
  };
  nextSlide = () => {
    if (this.state.currentSlide === this.props.data.length - 1) {
      this.setState({
        currentSlide: 0
      });
      return;
    }
    this.setState({
      currentSlide: this.state.currentSlide + 1
    });
    this.resetInterval();
  };

  prevSlide = () => {
    if (this.state.currentSlide === 0) {
      this.setState({
        currentSlide: this.props.data.length - 1
      });
      return;
    }
    this.setState({
      currentSlide: this.state.currentSlide - 1
    });
    this.resetInterval();
  };
  goToSlide = idx => {
    this.setState({
      currentSlide: idx
    });
    this.resetInterval();
  };
  render() {
    const {
      data,
      height,
      activeDotColor,
      dotColor,
      showDots,
      enableNextAndPrev
    } = this.props;
    return (
      <div className="swift-slider-container">
        <ul className="swift-slider-slides" style={{ height: height }}>
          {data.map((item, i) => (
            <Slide
              active={i === this.state.currentSlide}
              src={item.src}
              key={item.id}
            />
          ))}
        </ul>
        {showDots ? (
          <ul className="swift-slider-dots">
            {data.map((item, i) => (
              <Dot
                activeDotColor={activeDotColor}
                dotColor={dotColor}
                key={i}
                active={i === this.state.currentSlide}
                onClick={this.goToSlide}
                idx={i}
              />
            ))}
          </ul>
        ) : (
          ""
        )}
        {enableNextAndPrev ? (
          <Control onPressPrev={this.prevSlide} direction="prev" />
        ) : (
          ""
        )}
        {enableNextAndPrev ? (
          <Control onPressNext={this.nextSlide} direction="next" />
        ) : (
          ""
        )}
      </div>
    );
  }
}

Slider.propTypes = {
  data: PropTypes.array.isRequired,
  height: PropTypes.number,
  interval: PropTypes.number,
  activeDotColor: PropTypes.string,
  dotColor: PropTypes.string,
  showDots: PropTypes.bool,
  enableNextAndPrev: PropTypes.bool
};

Slider.defaultProps = {
  height: 450,
  activeDotColor: "#e8784e",
  interval: 5000,
  dotColor: "#909192",
  showDots: true,
  enableNextAndPrev: true
};
