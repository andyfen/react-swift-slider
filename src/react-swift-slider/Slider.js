import React from 'react';
import PropTypes from 'prop-types';
import Slide from './Slide';
import Control, { DIRECTION } from './Control';
import Dot from './Dot';
import styled from 'styled-components';

const SwiftSliderContainer = styled.div`
  position: relative;
  height: ${props => props.height || '450'}px;
  @media (max-width: 600px) {
    height: 250px;
  }
`;

const SwiftSliderSlides = styled.ul`
  position: relative;
  padding: 0;
  margin: 0;
  list-style: none;
  height: ${props => props.height || '450'}px;
  @media (max-width: 600px) {
    height: 250px;
  }
`;

const SwiftSliderDots = styled.ul`
  list-style: none;
  margin: 1.7rem 0;
  padding: 0;
  display: flex;
  flex-direction: row;
  justify-content: center;
  @media (max-width: 600px) {
    margin: 1rem 0;
  }
`;

export default class Slider extends React.Component {
  state = {
    currentSlide: 0,
  };

  componentDidMount() {
    this.setState({
      slideInterval: setInterval(this.nextSlide, this.props.interval),
    });
  }

  resetInterval = () => {
    clearInterval(this.state.slideInterval);
    this.setState({
      slideInterval: setInterval(this.nextSlide, this.props.interval),
    });
  };

  nextSlide = () => {
    if (this.state.currentSlide === this.props.data.length - 1) {
      this.setState({
        currentSlide: 0,
      });
      return;
    }
    this.setState({
      currentSlide: this.state.currentSlide + 1,
    });
    this.resetInterval();
  };

  prevSlide = () => {
    if (this.state.currentSlide === 0) {
      this.setState({
        currentSlide: this.props.data.length - 1,
      });
      return;
    }
    this.setState({
      currentSlide: this.state.currentSlide - 1,
    });
    this.resetInterval();
  };

  goToSlide = idx => {
    this.setState({
      currentSlide: idx,
    });
    this.resetInterval();
  };

  render() {
    const {
      data,
      height,
      activeDotColor,
      dotColor,
      showDots,
      enableNextAndPrev,
    } = this.props;
    return (
      <SwiftSliderContainer>
        <SwiftSliderSlides height={height}>
          {data.map((item, i) => (
            <Slide
              active={i === this.state.currentSlide}
              src={item.src}
              key={item.id}
            />
          ))}
        </SwiftSliderSlides>
        {showDots ? (
          <SwiftSliderDots>
            {data.map((item, i) => (
              <Dot
                activeDotColor={activeDotColor}
                dotColor={dotColor}
                key={i}
                active={i === this.state.currentSlide}
                onClick={this.goToSlide}
                idx={i}
              />
            ))}
          </SwiftSliderDots>
        ) : (
          ''
        )}
        {enableNextAndPrev ? (
          <Control onPressPrev={this.prevSlide} direction={DIRECTION.prev} />
        ) : (
          ''
        )}
        {enableNextAndPrev ? (
          <Control onPressNext={this.nextSlide} direction={DIRECTION.next} />
        ) : (
          ''
        )}
      </SwiftSliderContainer>
    );
  }
}

Slider.propTypes = {
  data: PropTypes.array.isRequired,
  height: PropTypes.number,
  interval: PropTypes.number,
  activeDotColor: PropTypes.string,
  dotColor: PropTypes.string,
  showDots: PropTypes.bool,
  enableNextAndPrev: PropTypes.bool,
};

Slider.defaultProps = {
  data: [],
  height: 450,
  activeDotColor: '#e8784e',
  interval: 5000,
  dotColor: '#909192',
  showDots: true,
  enableNextAndPrev: true,
};
