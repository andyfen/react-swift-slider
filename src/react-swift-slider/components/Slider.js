import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Slide from './Slide';
import Control from './Control';
import Dot from './Dot';
import { useInterval } from '../hooks/use-interval';
import {
  CSS_OVERRIDE,
  DEFAULT_PROPS,
  DIRECTION,
  SliderDataType,
} from '../config';
import SwiftSliderThumbnails from './Thumbnails';
import { getSlideId } from '../helpers';

const SwiftSliderWrapper = styled.div`
  overflow: hidden;
  height: 100vh;
  display: grid;
  grid-template-rows: 2fr 1fr;
  grid-gap: 10px;
`;

const SwiftSliderContainer = styled.div`
  position: relative;
  height: ${(props) => props.height || '450'}px;
  @media (max-width: 600px) {
    height: 250px;
  }
`;

const SwiftSliderSlides = styled.ul`
  position: relative;
  padding: 0;
  margin: 0;
  list-style: none;
  height: ${(props) => props.height || '450'}px;
  @media (max-width: 600px) {
    height: 250px;
  }
`;

const SwiftSliderDots = styled.ul`
  list-style: none;
  margin: 1.7rem 0;
  padding: 0;
  display: flex;
  flex-direction: row;
  justify-content: center;
  @media (max-width: 600px) {
    margin: 1rem 0;
  }
`;

function ReactSlider({
  data,
  height,
  activeDotColor,
  interval,
  dotColor,
  showDots,
  enableNextAndPrev,
  showThumbnails,
}) {
  const [currentSlide, setCurrentSlide] = useState(0);
  const [intervalId] = useInterval(() => {
    nextSlide();
  }, interval);

  useEffect(() => {
    data.forEach((img) => {
      const image = new Image();
      image.src = img.src;
    });
  }, [data]);

  const nextSlide = () => {
    if (currentSlide === data.length - 1) {
      return setCurrentSlide(0);
    }
    setCurrentSlide(currentSlide + 1);
  };

  const prevSlide = () => {
    if (currentSlide === 0) {
      return setCurrentSlide(data.length - 1);
    }
    setCurrentSlide(currentSlide - 1);
    clearInterval(intervalId);
  };

  const goToSlide = (idx) => {
    setCurrentSlide(idx);
    clearInterval(intervalId);
  };

  return (
    <SwiftSliderWrapper>
      <SwiftSliderContainer
        height={height}
        className={CSS_OVERRIDE.swiftSliderContainerClass}
      >
        <SwiftSliderSlides
          height={height}
          className={CSS_OVERRIDE.swiftSliderSlidesClass}
        >
          {data.map((item, i) => (
            <Slide
              active={i === currentSlide}
              src={item.src}
              key={item.id}
              id={getSlideId(item.id)}
            />
          ))}
        </SwiftSliderSlides>
        {showDots ? (
          <SwiftSliderDots className={CSS_OVERRIDE.swiftSliderDotsClass}>
            {data.map((item, i) => (
              <Dot
                activeDotColor={activeDotColor}
                dotColor={dotColor}
                key={i}
                active={i === currentSlide}
                onClick={goToSlide}
                idx={i}
              />
            ))}
          </SwiftSliderDots>
        ) : (
          ''
        )}
        {enableNextAndPrev ? (
          <Control
            onPressPrev={prevSlide}
            direction={DIRECTION.prev}
            height={height}
          />
        ) : (
          ''
        )}
        {enableNextAndPrev ? (
          <Control
            onPressNext={nextSlide}
            direction={DIRECTION.next}
            height={height}
          />
        ) : (
          ''
        )}
      </SwiftSliderContainer>

      <SwiftSliderThumbnails showThumbnails={showThumbnails} data={data} />
    </SwiftSliderWrapper>
  );
}

ReactSlider.propTypes = {
  activeDotColor: PropTypes.string,
  data: SliderDataType,
  dotColor: PropTypes.string,
  enableNextAndPrev: PropTypes.bool,
  height: PropTypes.number,
  interval: PropTypes.number,
  showDots: PropTypes.bool,
  showThumbnails: PropTypes.bool,
};

ReactSlider.defaultProps = {
  ...DEFAULT_PROPS,
};

export default ReactSlider;
