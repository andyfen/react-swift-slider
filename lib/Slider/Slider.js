import React from 'react';
import styles from './styles.css';

import Slide from '../Slide';
import Control from '../Control';
import Dot from '../Dot';

class Slider extends React.Component {
  static propTypes = {
    data: React.PropTypes.array.isRequired,
    height: React.PropTypes.number,
    interval: React.PropTypes.number,
    activeDotColor: React.PropTypes.string,
    dotColor: React.PropTypes.string
  };
  static defaultProps = {
    height: 450,
    activeDotColor: '#e8784e',
    interval: 5000,
    dotColor: '#909192'
  };
  constructor(props){
    super(props);
    this.state = {
      currentSlide: 0,
    }
    this.slideInterval = setInterval(this.nextSlide, this.props.interval);
  }
  resetInterval = () => {
    clearInterval(this.slideInterval)
    this.slideInterval = setInterval(this.nextSlide, this.props.interval);
  }
  nextSlide = () => {
    if(this.state.currentSlide === this.props.data.length-1){
      this.setState({
        currentSlide: 0
      })
      return;
    }
    this.setState({
      currentSlide: this.state.currentSlide +1
    })
    this.resetInterval();
  }

  prevSlide = () => {
    if(this.state.currentSlide === 0){
      this.setState({
        currentSlide: this.props.data.length-1
      })
      return;
    }
    this.setState({
      currentSlide: this.state.currentSlide -1
    })
    this.resetInterval();
  }
  goToSlide = (idx) => {
    this.setState({
      currentSlide: idx
    })
    this.resetInterval();
  }
  render(){
    const { data, height, activeDotColor, dotColor } = this.props;
    return(
      <div className={styles.container} style={{ height }}>

        <ul className={styles.slides} style={{ height }}>
          {data.map((item, i) =>
            <Slide active={i === this.state.currentSlide} src={item.src} key={item.id}/>
          )}
        </ul>
        <ul className={styles.dots}>
          {data.map((item, i) =>
            <Dot
              activeDotColor={activeDotColor}
              dotColor={dotColor}
              key={i} active={i === this.state.currentSlide}
              onClick={this.goToSlide}
              idx={i}
            />
          )}
        </ul>
        <Control
          onPressPrev={this.prevSlide}
          direction="prev"/>
        <Control
          onPressNext={this.nextSlide}
          direction="next"/>
      </div>
    )
  }
}

export default Slider;
