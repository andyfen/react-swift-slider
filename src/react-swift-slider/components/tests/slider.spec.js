/**
 * Created by Muzikayise Flynn Buthelezi (zuluCoda) on 2017/11/18.
 * Copyright mfbproject.co.za - muzi@mfbproject.co.za
 * Copyright zulucoda - mfbproject
 */
import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import Slider from '../Slider';
import 'jest-dom/extend-expect';

describe('Slider - Unit Test', () => {
  afterEach(cleanup);

  const data = [
    {
      id: '1',
      src: 'https://picjumbo.imgix.net/HNCK9192.jpg?q=40&w=1650&sharp=30',
    },
    {
      id: '2',
      src: 'https://picjumbo.imgix.net/HNCK3313.jpg?q=40&w=1650&sharp=30',
    },
    {
      id: '3',
      src: 'https://picjumbo.imgix.net/HNCK4330.jpg?q=40&w=1650&sharp=30',
    },
    {
      id: '4',
      src: 'https://picjumbo.imgix.net/HNCK0180.jpg?q=40&w=1650&sharp=30',
    },
    {
      id: '5',
      src: 'https://picjumbo.imgix.net/HNCK3286.jpg?q=40&w=1650&sharp=30',
    },
  ];

  it('should render slider', () => {
    const { container } = render(<Slider data={data} />);
    expect(container.querySelector('div > ul').childElementCount).toEqual(5);
    expect(
      container.querySelector('div >:nth-child(2)').childElementCount,
    ).toEqual(5);
    expect(container.querySelector('div >:nth-child(3)')).toBeTruthy();
    expect(container.querySelector('div >:nth-child(2)')).toBeTruthy();
  });

  it('should go to position 4 when on the first slide when clicking on next', () => {
    const { container } = render(<Slider data={data} />);

    expect(container.querySelector('div > ul > li')).toHaveStyle(
      'visibility : visible',
    );
    expect(container.querySelector('div > ul > :nth-child(5)')).toHaveStyle(
      'visibility : hidden',
    );

    const prevButton = container.querySelector('div >:nth-child(3)');
    fireEvent.click(prevButton);

    expect(container.querySelector('div > ul > li')).toHaveStyle(
      'visibility : hidden',
    );
    expect(container.querySelector('div > ul > :nth-child(5)')).toHaveStyle(
      'visibility : visible',
    );
  });

  it('should go to previous slide', () => {
    const { container } = render(<Slider data={data} />);
    const prevButton = container.querySelector('div >:nth-child(3)');
    fireEvent.click(prevButton);

    expect(container.querySelector('div > ul > li')).toHaveStyle(
      'visibility : hidden',
    );
    expect(container.querySelector('div > ul > :nth-child(4)')).toHaveStyle(
      'visibility : hidden',
    );
    expect(container.querySelector('div > ul > :nth-child(5)')).toHaveStyle(
      'visibility : visible',
    );

    fireEvent.click(prevButton);

    expect(container.querySelector('div > ul > li')).toHaveStyle(
      'visibility : hidden',
    );
    expect(container.querySelector('div > ul > :nth-child(4)')).toHaveStyle(
      'visibility : visible',
    );
    expect(container.querySelector('div > ul > :nth-child(5)')).toHaveStyle(
      'visibility : hidden',
    );
  });

  it('should go to next slide', () => {
    const { container } = render(<Slider data={data} />);
    expect(container.querySelector('div > ul > li')).toHaveStyle(
      'visibility : visible',
    );
    expect(container.querySelector('div > ul > :nth-child(2)')).toHaveStyle(
      'visibility : hidden',
    );

    const nextButton = container.querySelector('div >:nth-child(4)');
    fireEvent.click(nextButton);

    expect(container.querySelector('div > ul > li')).toHaveStyle(
      'visibility : hidden',
    );
    expect(container.querySelector('div > ul > :nth-child(2)')).toHaveStyle(
      'visibility : visible',
    );
  });

  it('should go to position 0 when on the last slide when clicking on next', () => {
    const { container } = render(<Slider data={[data[0], data[1]]} />);
    expect(container.querySelector('div > ul > li')).toHaveStyle(
      'visibility : visible',
    );
    expect(container.querySelector('div > ul > :nth-child(2)')).toHaveStyle(
      'visibility : hidden',
    );

    const nextButton = container.querySelector('div >:nth-child(4)');
    fireEvent.click(nextButton);

    expect(container.querySelector('div > ul > li')).toHaveStyle(
      'visibility : hidden',
    );
    expect(container.querySelector('div > ul > :nth-child(2)')).toHaveStyle(
      'visibility : visible',
    );

    fireEvent.click(nextButton);

    expect(container.querySelector('div > ul > li')).toHaveStyle(
      'visibility : visible',
    );
    expect(container.querySelector('div > ul > :nth-child(2)')).toHaveStyle(
      'visibility : hidden',
    );
  });

  it('should go slide when clicking on a dot', () => {
    const { container } = render(<Slider data={data} />);
    const dotUl = container.querySelector('div >:nth-child(2)');

    expect(dotUl.querySelector(':nth-child(1)')).toHaveStyle(
      'background : #e8784e',
    );
    expect(dotUl.querySelector(':nth-child(3)')).toHaveStyle(
      'background : #909192',
    );

    fireEvent.click(dotUl.querySelector(':nth-child(3)'));

    expect(dotUl.querySelector(':nth-child(1)')).toHaveStyle(
      'background : #909192',
    );
    expect(dotUl.querySelector(':nth-child(3)')).toHaveStyle(
      'background : #e8784e',
    );
  });

  it('should not show dots', () => {
    const { container } = render(<Slider data={data} showDots={false} />);
    expect(container.querySelectorAll('ul').length).toBe(1);
  });

  it('should not show next and previous', () => {
    const { container } = render(
      <Slider data={data} enableNextAndPrev={false} />,
    );
    expect(container.querySelector('div >:nth-child(3)')).toBeFalsy();
    expect(container.querySelector('div >:nth-child(4)')).toBeFalsy();
  });
});
