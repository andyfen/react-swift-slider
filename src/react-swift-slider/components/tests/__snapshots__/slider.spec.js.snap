// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider - Unit Test css override handles should get all dots by using className - this will allow for default styling override 1`] = `
NodeList [
  <li
    class="sc-eCssSg ihWeoK swift-slider-dot"
  />,
  <li
    class="sc-eCssSg ihWeoK swift-slider-dot"
  />,
  <li
    class="sc-eCssSg ihWeoK swift-slider-dot"
  />,
  <li
    class="sc-eCssSg ihWeoK swift-slider-dot"
  />,
]
`;

exports[`Slider - Unit Test css override handles should get all slides by using className - this will allow for default styling override 1`] = `
NodeList [
  <li
    class="sc-bdfBwQ esexun swift-slider-slide"
  />,
  <li
    class="sc-bdfBwQ kQtNob swift-slider-slide"
  />,
  <li
    class="sc-bdfBwQ iGSsDB swift-slider-slide"
  />,
  <li
    class="sc-bdfBwQ bfqwzu swift-slider-slide"
  />,
]
`;

exports[`Slider - Unit Test css override handles should get slider container by using className - this will allow for default styling override 1`] = `
<div
  class="sc-fubCfw grUhyC swift-slider-container"
  height="450"
>
  <ul
    class="sc-pFZIQ dVEDgL swift-slider-slides"
    height="450"
  >
    <li
      class="sc-bdfBwQ jFksPw swift-slider-active-slide"
    />
    <li
      class="sc-bdfBwQ esexun swift-slider-slide"
    />
    <li
      class="sc-bdfBwQ kQtNob swift-slider-slide"
    />
    <li
      class="sc-bdfBwQ iGSsDB swift-slider-slide"
    />
    <li
      class="sc-bdfBwQ bfqwzu swift-slider-slide"
    />
  </ul>
  <ul
    class="sc-jrAGrp iiYTZC swift-slider-dots"
  >
    <li
      class="sc-eCssSg bRDNPX swift-slider-active-dot"
    />
    <li
      class="sc-eCssSg ihWeoK swift-slider-dot"
    />
    <li
      class="sc-eCssSg ihWeoK swift-slider-dot"
    />
    <li
      class="sc-eCssSg ihWeoK swift-slider-dot"
    />
    <li
      class="sc-eCssSg ihWeoK swift-slider-dot"
    />
  </ul>
  <div
    class="sc-gsTCUz sc-dlfnbm gLXrjP jAVIW swift-slider-prev-btn"
    height="450"
  />
  <div
    class="sc-gsTCUz sc-hKgILt gLXrjP kcsjwH swift-slider-next-btn"
    height="450"
  />
</div>
`;

exports[`Slider - Unit Test css override handles should get slider dots container by using className - this will allow for default styling override 1`] = `
<ul
  class="sc-jrAGrp iiYTZC swift-slider-dots"
>
  <li
    class="sc-eCssSg bRDNPX swift-slider-active-dot"
  />
  <li
    class="sc-eCssSg ihWeoK swift-slider-dot"
  />
  <li
    class="sc-eCssSg ihWeoK swift-slider-dot"
  />
  <li
    class="sc-eCssSg ihWeoK swift-slider-dot"
  />
  <li
    class="sc-eCssSg ihWeoK swift-slider-dot"
  />
</ul>
`;

exports[`Slider - Unit Test css override handles should get slider slides container by using className - this will allow for default styling override 1`] = `
<ul
  class="sc-pFZIQ dVEDgL swift-slider-slides"
  height="450"
>
  <li
    class="sc-bdfBwQ jFksPw swift-slider-active-slide"
  />
  <li
    class="sc-bdfBwQ esexun swift-slider-slide"
  />
  <li
    class="sc-bdfBwQ kQtNob swift-slider-slide"
  />
  <li
    class="sc-bdfBwQ iGSsDB swift-slider-slide"
  />
  <li
    class="sc-bdfBwQ bfqwzu swift-slider-slide"
  />
</ul>
`;

exports[`Slider - Unit Test css override handles should get the active dot by using className - this will allow for default styling override 1`] = `
NodeList [
  <li
    class="sc-eCssSg bRDNPX swift-slider-active-dot"
  />,
]
`;

exports[`Slider - Unit Test css override handles should get the active slide by using className - this will allow for default styling override 1`] = `
NodeList [
  <li
    class="sc-bdfBwQ jFksPw swift-slider-active-slide"
  />,
]
`;

exports[`Slider - Unit Test css override handles should get the next btn by using className - this will allow for default styling override 1`] = `
<div
  class="sc-gsTCUz sc-hKgILt gLXrjP kcsjwH swift-slider-next-btn"
  height="450"
/>
`;

exports[`Slider - Unit Test css override handles should get the previous btn by using className - this will allow for default styling override 1`] = `
<div
  class="sc-gsTCUz sc-dlfnbm gLXrjP jAVIW swift-slider-prev-btn"
  height="450"
/>
`;
